pkg/azurefile/controllerserver.go

```diff
@@ -212,7 +215,51 @@ func (d *Driver) ControllerUnpublishVolume(ctx context.Context, req *csi.Control
 
 // CreateSnapshot create a snapshot (todo)
 func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequest) (*csi.CreateSnapshotResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "")
+	klog.V(2).Infof("CreateSnapshot called with request %v", *req)
+
+	sourceVolumeID := req.GetSourceVolumeId()
+	snapshotName := req.Name
+	if len(snapshotName) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Snapshot name must be provided")
```

Comment:
Snapshot name is empty

Reply:
If it is not checked here, the related sanity test needs to be skipped.